File: GXpert/GXpert.Web/Modules/QuestionBank/Question/QuestionForm.cs
using Serenity.ComponentModel;
using System;

namespace GXpert.QuestionBank.Forms;

[FormScript("QuestionBank.Question")]
[BasedOnRow(typeof(QuestionRow), CheckNames = true)]
public class QuestionForm
{
    [Category("Question")]
    [HtmlContentEditor(Rows = 4), CssClass("fact")]
    public string QuestionText { get; set; }
    [HtmlContentEditor(Rows = 4), CssClass("fact")]
    public string Explaination { get; set; }
    
    [Category("Question Attributes")]
    [HalfWidth, CssClass("fact")]
    public bool IsSubjective { get; set; }
    [HalfWidth, CssClass("fact")]
    public short EQuestionType { get; set; }
    [HalfWidth, CssClass("fact")]
    public short EDifficultyLevel { get; set; }
    [HalfWidth, CssClass("fact")]
    public int QuestionCommonDataId { get; set; }
    [HalfWidth, CssClass("fact")]
    public float QuestionCommonDataSortOrder { get; set; }
    [HalfWidth, CssClass("fact")]
    public int ClassId { get; set; }
    [HalfWidth, CssClass("fact")]
    public int SubjectId { get; set; }
    [HalfWidth, CssClass("fact")]
    public int TopicId { get; set; }
    [HalfWidth, CssClass("fact")]
    public int BloomsIndex { get; set; }
    [HalfWidth, CssClass("fact")]
    public float BloomsWeightage { get; set; }
    [Category("QuestionOption"), CssClass("fact")]
    [QuestionOptionEditor]
    public List<QuestionOptionRow> QuestionOptions { get; set; }
}

File: GXpert/GXpert.Web/Modules/ServerTypes/QuestionBank/QuestionForm.ts
import { HtmlContentEditor, BooleanEditor, EnumEditor, ServiceLookupEditor, DecimalEditor, LookupEditor, PrefixedContext, initFormType } from "@serenity-is/corelib";
import { EDifficultyLevel } from "../Web/Enums.EDifficultyLevel";
import { EQuestionType } from "../Web/Enums.EQuestionType";

export interface QuestionForm {
    QuestionText: HtmlContentEditor;
    Explaination: HtmlContentEditor;
    IsSubjective: BooleanEditor;
    EQuestionType: EnumEditor;
    EDifficultyLevel: EnumEditor;
    QuestionCommonDataId: ServiceLookupEditor;
    QuestionCommonDataSortOrder: DecimalEditor;
    ClassId: LookupEditor;
    SubjectId: LookupEditor;
    TopicId: LookupEditor;
    BloomsIndex: LookupEditor;
    BloomsWeightage: DecimalEditor;
}

export class QuestionForm extends PrefixedContext {
    static readonly formKey = 'QuestionBank.Question';
    private static init: boolean;

    constructor(prefix: string) {
        super(prefix);

        if (!QuestionForm.init)  {
            QuestionForm.init = true;

            var w0 = HtmlContentEditor;
            var w1 = BooleanEditor;
            var w2 = EnumEditor;
            var w3 = ServiceLookupEditor;
            var w4 = DecimalEditor;
            var w5 = LookupEditor;

            initFormType(QuestionForm, [
                'QuestionText', w0,
                'Explaination', w0,
                'IsSubjective', w1,
                'EQuestionType', w2,
                'EDifficultyLevel', w2,
                'QuestionCommonDataId', w3,
                'QuestionCommonDataSortOrder', w4,
                'ClassId', w5,
                'SubjectId', w5,
                'TopicId', w5,
                'BloomsIndex', w5,
                'BloomsWeightage', w4
            ]);
        }
    }
}

[EQuestionType, EDifficultyLevel]; // referenced types